modules/imgcodecs/include/opencv2/imgcodecs.hpp

```diff
@@ -322,6 +324,59 @@ CV_EXPORTS_W bool haveImageReader( const String& filename );
  */
 CV_EXPORTS_W bool haveImageWriter( const String& filename );
 
+class CV_EXPORTS ImageCollection {
+    String m_filename;
+    int m_flags;
+    size_t m_size;
+    std::vector<Mat> m_data;
+
+private:
+    explicit ImageCollection(String filename, int flags, size_t size);
+
+public:
+
+    struct Iterator;
+
+    CV_WRAP size_t nimages() const;
+    Mat operator[](int index);
+    CV_WRAP Mat at(int index);
+    CV_WRAP void free(int index);
+    CV_WRAP Iterator begin();
+    CV_WRAP Iterator end();
+
+    CV_WRAP static ImageCollection fromMultiPageImage(const std::string& img, int flags);
+
+    struct Iterator
```

Comment:
Please, hide the implementation details as well. Let's keep the headers as clean as possible